src/fields/fq2.rs

```diff
@@ -255,15 +256,14 @@ impl FieldElement for Fq2 {
         // "High-Speed Software Implementation of the Optimal Ate Pairing
         // over Barreto–Naehrig Curves"; Algorithm 8
 
-        (self
-            .c0
-            .cpu_mul(self.c0)
-            .cpu_sub((self.c1.cpu_mul(self.c1)).cpu_mul(fq_non_residue())))
-        .inverse()
-        .map(|t| Fq2 {
-            c0: self.c0.cpu_mul(t),
-            c1: (self.c1.cpu_mul(t)).cpu_neg(),
-        })
+        self.c0
+            .mul(self.c0)
+            .sub((self.c1.mul(self.c1)).mul(fq_non_residue()))
+            .inverse_unconstrained()
```

Comment:
Should we gate this behind a `zkvm` feature flag given that we're changing it from `cpu_mul` to zkvm `mul`, and `inverse_unconstrained`

Reply:
cpu_mul -> mul
This was causing the cycle count to go up by like 10% per pairing. mul and sub already check which environment we're running the patch in before correctly routing to either cpu_mul or the syscall mul -- there's no reason to force cpu_mul at this level.

also inverse_unconstrained automatically routes to inverse if the target os is not zkvm